canvas_sdk/commands/base.py

```diff
@@ -66,23 +70,25 @@ def command_schema(cls) -> dict:
 
     def originate(self) -> Effect:
```

Comment:
We should run the pydantic validators when these effect methods are called rather than on assignment. This should allow us to set the attributes individually instead of only in the constructor.
Example:
```python
plan = PlanCommand()
plan.narrative = "This is my narrative"
plan.originate()  # <-- only validate here
```

Reply:
good idea. want me to tackle that?

Reply:
Sure, once this gets merged. We should chat about other priorities soon too.